<!DOCTYPE html>
<html lang="en">

<head>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
  <meta charset="utf-8" />
  <title>E. coli</title>
  <style>
    :root {
      --slider-pip-left-value: 0%;
    }

    body {
      margin: 0;
      font-family: system-ui, -apple-system, sans-serif;
    }

    canvas {
      display: block;
      overflow: hidden;
      position: relative;
    }

    img {
      display: block;
      max-width: 100%;
    }

    article {
      max-width: 38em;
      font-size: 18px;
      line-height: 1.3;
      margin: 18px 18px 0;
    }

    h1 {
      font-size: clamp(2rem, 5vw, 6rem);
      font-weight: 800;
      margin: 0.3em auto;
      max-width: calc(100vw - 36px);
      line-height: 1.05;
    }

    h2 {
      font-size: inherit;
      margin: 0;
    }

    .heroContainer {
      position: relative;
      margin: 1em auto 0;
      width: calc(100vw - 32px);
    }

    .heroContainer canvas {
      background: url("./ecoliSimulation/images/Background.png") #010010;
      background-size: 100% 100%;
    }

    .heroGif {
      position: absolute;
      top: 16px;
      right: 16px;
      border-radius: 14px;
      overflow: hidden;
    }

    .feedButton {
      width: 100px;
      height: 70px;
      position: absolute;
      bottom: 16px;
      right: 16px;
      border-radius: 14px;
      display: flex;
      align-items: center;
      justify-content: center;
      cursor: pointer;
      background: rgba(255, 255, 255, 0.4);
      backdrop-filter: blur(18px);
      font-family: system-ui, -apple-system, sans-serif;
      font-weight: 600;
      letter-spacing: 0.5px;
      color: rgba(255, 255, 255, 0.9);
      user-select: none;
      transition: transform, background 0.2s ease;
    }

    .feedButton:hover {
      transform: translate(0, -1px) scale(1.01);
      background: rgba(255, 255, 255, 0.5);
      color: #fff;
    }

    .feedButton:active {
      transform: translate(0, 1px) scale(0.99);
      transition-duration: 0s;
      background: rgba(255, 255, 255, 0.25);
      color: rgba(255, 255, 255, 0.6);
    }

    .sliderAndPip {
      position: relative;
      width: calc(100vw - 32px);
      height: 64px;
      margin: 8px auto 0;
    }

    .sliderAndPip canvas {
      cursor: pointer;
    }

    #debug1,
    #debug2 {
      position: fixed;
      bottom: 0;
      left: 0;
      background: black;
      color: white;
    }
  </style>
</head>

<body>
  <h1> The baffling intelligence of a single cell: the story of E. coli chemotaxis </h1>
  <div class="heroContainer">
    <div class="heroCanvasContainer"></div>
    <div class="heroGif"></div>
    <div class="feedButton" id="feedButton">FEED</div>
  </div>
  <div class="sliderAndPip"></div>
  <article>
    <h2>By James Somers and Edwin Morris</h2>
    <p>
      <span>I want to tell the story of a beautiful phenomenon in biology. In some sense it&rsquo;s the prototype of
        much of the activity of life. The phenomenon is the way in which an individual cell of </span><span>E.
        coli</span><span>&nbsp;forages for nutrients. This process, known as &ldquo;chemotaxis&rdquo;&mdash;the
        &ldquo;chemo-&rdquo; for chemical and the &ldquo;taxis&rdquo; from the Greek &tau;&#940;&xi;&iota;&sigmaf;, for
        tactics&mdash;is intelligence in one of its most elemental forms. An individual </span><span>E.
        coli</span><span>&nbsp;has no brain, obviously, and is even many orders of magnitude simpler than a human cell,
        and yet already it possesses something like a sense of smell, drive, even a memory. Chemotaxis recasts
      </span><span>E. coli</span><span>&nbsp;not as some aimless gut-pest but rather as an exquisitely sophisticated
      </span><span><a
          href="https://www.google.com/url?q=https://yalebooks.yale.edu/book/9780300167849/wetware/%23:~:text%3DThis%2520book%2520offers%2520a%2520startling,is%2520a%2520form%2520of%2520computation.&amp;sa=D&amp;source=editors&amp;ust=1691027434677832&amp;usg=AOvVaw2VnPLTTSzSRt9dFNZbxDM9">physical
          computer</a></span><span>.</span>
    </p>
    <p>
      <span>I&rsquo;m also telling this story because </span><span><a
          href="https://www.google.com/url?q=https://jsomers.net/i-should-have-loved-biology&amp;sa=D&amp;source=editors&amp;ust=1691027434678206&amp;usg=AOvVaw2HTu6Vf93Lwr2dYM4A1LCC">I
          never liked the way biology was taught in high school</a></span><span>. It was too much about the names of
        things. A subject so vast is spoiled by a textbook, which can only point, rotely, at the endless parade of
        stuff-there-is-to-know. It&rsquo;s better approached in narrow, deep slices. Chemotaxis is a great slice:
      </span><span>it&rsquo;s a triumph of systems biology&mdash;we understand it holistically but also in fine detail
        at almost every level. It acquaints you with many of the most important motifs in biology, including the way in
        which protein structure determines function; how membranes control the information flow into cells; and how
        chemical modifications store and communicate state. It involves one of the most sophisticated and beautiful
        pieces of molecular nanotechnology, the flagellar motor. And it helps give an intuition for how a bag of
        unthinking chemicals could possibly give rise to a being.</span>
    </p>
    <p>
      <span>The 30,000-foot view</span><span><br />The basic idea is this: </span><span>E.
        coli</span><span>&nbsp;&ldquo;smells&rdquo; chemicals it&rsquo;s attracted to with a set of nose-like receptors
        on its front. If attractant is present, it uses its flagellar tails to swim forward&mdash;this is known as a
        &ldquo;run&rdquo;&mdash;and otherwise it spins in a random direction (a &ldquo;tumble&rdquo;). By running when
        the getting is good and tumbling when it isn&rsquo;t, the </span><span>E. coli</span><span>&nbsp;takes a
        meandering path toward the attractant.</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image5.png" />
    </p>
    <p>
      <span>(Adapted from La Vecchia, </span><span><a
          href="https://www.google.com/url?q=http://www.diva-portal.org/smash/get/diva2:430333/FULLTEXT01.pdf&amp;sa=D&amp;source=editors&amp;ust=1691027434679093&amp;usg=AOvVaw1QM34q00bv1Cye2NHj9SsE">&ldquo;Bacterial
          chemotaxis in non-homogeneous shear flow&rdquo;</a></span><span>)</span>
    </p>
    <p>
      <span>A little more detail now: there are half a dozen or so rotors on the </span><span>E.
        coli</span><span>&rsquo;s body, each controlling a long whip-like tail that flows behind it. When all the rotors
        are spinning in the same direction, the tails join together into a coil that torques the cell forward into a
        run. When even one rotor is spinning against the others, the coil unbundles and </span><span>E.
        coli</span><span>&nbsp;spins into a tumble.</span>
    </p>
    <p>
      <span>But how do the rotors know which way to spin? The connection between the nose and the motor is given by a
        signaling molecule known as CheyY (pronounced like &ldquo;KEY-why&rdquo;). CheY is constantly bouncing around in
        the cytoplasm of the </span><span>E. coli</span><span>. In its normal state, it binds to those receptors in the
        nose. But when a receptor encounters attractant, CheY gets chemically tagged and its shape changes so that now
        it has a strong affinity for the rotors. The modified CheY-p, as it&rsquo;s known, binds to the rotor and, via a
        very cool mechanism that we&rsquo;ll get into later, reverses the flagellum&rsquo;s direction of spin.</span>
    </p>
    <p>
      <span>You can see this process in action in the interactive illustration. Try adding some attractant.</span>
    </p>
    <p>
      <span>[that same </span><span><a
          href="https://www.google.com/url?q=https://ehmorris.github.io/ecoli-chemotaxis/&amp;sa=D&amp;source=editors&amp;ust=1691027434679850&amp;usg=AOvVaw2pkV3FvCfSKpCWHOYJ9Xzj">interactive
          illustration</a></span><span>]</span>
    </p>
    <p>
      <span>The default behavior of the rotors is to spin clockwise, which makes the </span><span>E.
        coli</span><span>&nbsp;tumble. When attractant is detected, the CheY gets
        transformed&mdash;&ldquo;phosphorylated&rdquo;&mdash;into a new shape that binds to the rotors. You can see that
        the cell is extremely responsive to attractant: just a tiny bit causes a flurry of phosphorylation of CheY into
        CheY-p. When enough CheY-ps attach to a rotor, it starts spinning counterclockwise. When all the rotors spin
        that way, the </span><span>E. coli</span><span>&nbsp;runs.</span>
    </p>
    <p>
      <span>Given that setup, it may seem like the </span><span>E. coli</span><span>&nbsp;should be tumbling most of the
        time. But in real life, </span><span>E. coli</span><span>&nbsp;are near attractant often enough to be running a
        lot, and once they start a run, they tend to stay in it awhile: the chemical modifications that revert the
        rotors to their default states are tuned so that runs last on average about a second, or ten times longer than
        an average tumble. This ratio of run duration to tumble duration is crucial, as it defines the bacterium&rsquo;s
        balance of &ldquo;exploitation&rdquo; versus &ldquo;exploration&rdquo;: if runs lasted too long, </span><span>E.
        coli</span><span>&nbsp;would range too widely and zip past their food; too short, and they&rsquo;d likely never
        find it in the first place.</span>
    </p>
    <p>
      <span>Why do you need all this complexity? You could imagine a system in which the motors themselves responded
        directly to attractant. We&rsquo;ll see later on that the stream of CheYs acts as a kind of adaptable, tunable
        chemical amplifier. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.ks.uiuc.edu/Research/chemotaxis/%23:~:text%3DIn%2520particular%252C%2520bacterial,per%2520cell%2520volume!&amp;sa=D&amp;source=editors&amp;ust=1691027434680875&amp;usg=AOvVaw3Ucae8bgqOXXuWUxzs3zT-">Bacterial
          cells can amplify</a></span><span>&nbsp;signals more than 50-fold; that is to say, a 2% change in receptor
        occupancy can bring about a 100% change in the output of the system at the flagellar motors. This feature allows
        cells to sense minute changes in concentration&mdash;less than three molecules per cell volume!&rdquo;</span>
    </p>
    <p>
      <span>The story gets more complicated: adaptation</span><span><br />If the system were as described above, then
        the </span><span>E. coli</span><span>&nbsp;wouldn&rsquo;t have much dynamic range. Imagine: if a mere three
        molecules of attractant can set off a flood of CheY phosphorylation&mdash;enough to make the cell switch from
        tumbling to running&mdash;then how could it possibly react any more aggressively to yet
      </span><span>more</span><span>&nbsp;attractant? It&rsquo;s way too keyed up.</span>
    </p>
    <p>
      <span>In reality, the </span><span>E. coli</span><span>&nbsp;is equally responsive across </span><span><a
          href="https://www.google.com/url?q=https://youtu.be/cT855rpX8bc?t%3D58&amp;sa=D&amp;source=editors&amp;ust=1691027434681554&amp;usg=AOvVaw3GXW5WNNvtCtbU0HUN84tl">five
          orders of magnitude</a></span><span>&nbsp;of attractant concentration. The cell learns to treat whatever
        concentration it stumbles into as the new normal, so that the slightest increase triggers the same
        hypersensitive response as always.</span>
    </p>
    <p>
      <span>The mechanism powering this adaptation is extremely clever. Each receptor is equipped with
        &ldquo;struts&rdquo; with little closed pockets in them. (There are some detailed illustrations of this later.)
        When the receptor is bound to attractant, its struts change shape so that these pockets open up, and become the
        targets for little molecules known as methyl groups. Methyl groups are ubiquitous in biochemistry: gene
        transcription, for instance, is heavily regulated by the binding of methyl groups to structural proteins your
        DNA strands coil around, called histones; the &ldquo;methylated&rdquo; histone can kink the DNA strand into or
        out of view of your transcription machinery, turning it on or off.</span>
    </p>
    <p>
      <span>In this case, methylation serves to fill up the strut&rsquo;s pockets, causing it to become more rigid. With
        more rigid struts the receptor&rsquo;s signaling power is dampened: it takes more attractant to elicit the same
        response. &nbsp;Because there are many methylation sites per strut and many struts per receptor, there&rsquo;s a
        wide range of possible dampening values&mdash;as if those pockets were really the holes of an elaborate wind
        instrument. This wide dynamic range is what allows the bacteria not just to find a favorable environment but to
        keenly and speedily nose its way up a chemical gradient. No wonder a similar mechanism is used by cells in your
        immune system to track and hunt down invaders.</span>
    </p>
    <p>
      <span>Methylation of the receptors gives </span><span>E. coli</span><span>&nbsp;a &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2899694/%23:~:text%3DThe%2520receptor%2520methylation%2520level%2520also%2520provides%2520a%2520simple%2520chemical%2520memory%2520used%2520to%2520ascertain%2520whether%2520the%2520current%2520direction%2520of%2520swimming%2520is%2520favorable%2520or%2520unfavorable.&amp;sa=D&amp;source=editors&amp;ust=1691027434682346&amp;usg=AOvVaw3wFrTZhiRLw97LT8AnyaVv">simple
          chemical memory</a></span><span>.</span><span>&rdquo; This is a powerful and somewhat profound idea:
        individual bacteria can model their environment and remember important features of it by encoding that
        information in internal chemical modifications. </span><span>E. coli</span><span>&nbsp;&ldquo;knows&rdquo; the
        precise concentration of attractant in its surroundings going back </span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2899694&amp;sa=D&amp;source=editors&amp;ust=1691027434682737&amp;usg=AOvVaw0z4xCGPDADU7sEb9uo4-7g">several
          seconds</a></span><span>; that helps it determine whether it&rsquo;s swimming in a good or bad direction.
        Which is not that different in principle from what brains do. In fact one reason that it requires an artificial
        neural network of about a </span><span><a
          href="https://www.google.com/url?q=https://www.sciencedirect.com/science/article/pii/S0896627321005018&amp;sa=D&amp;source=editors&amp;ust=1691027434683037&amp;usg=AOvVaw0on3pPTGGoxEnj7LLEoiP7">thousand</a></span><span>&nbsp;elements
        just to model the computational capabilities of a single real neuron is that the real neuron stores so much
        &ldquo;state&rdquo; in its internal chemistry.</span>
    </p>
    <p>
      <span>The full picture: a complex signaling network</span>
    </p>
    <p>
      <span>[</span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3DLgPDOSou1tw&amp;sa=D&amp;source=editors&amp;ust=1691027434683443&amp;usg=AOvVaw0BcFI6z4ix_QZoQ7BL0wWo">Chemotaxis:
          Molecular Events</a></span><span>&nbsp;YouTube video</span><span>]</span>
    </p>
    <p>
      <span>The video above is a great overview of </span><span>E. coli</span><span>&nbsp;chemotaxis. It layers in even
        more detail, including not just the proteins that phosphorylate CheY but those that
      </span><span>de</span><span>phosphorylate it; and not just the proteins that methylate the receptors but those
        that </span><span>de</span><span>methylate it. What you come to see is that these doers and undoers define a
        sort of equilibrated circuit whose activity can be conveniently dialed up or down.</span>
    </p>
    <p>
      <span>Dennis Bray describes these sorts of circuits nicely in his book, </span><span>Wetware: A Computer in Every
        Living Cell</span><span>:</span>
    </p>
    <p>
      <span>In a typical signaling pathway, proteins are continually being modified and demodified. Kinases and
        phosphatases work ceaselessly like ants in a nest, adding phosphate groups to proteins and removing them again.
        It seems a pointless exercise, especially when you consider that each cycle of addition and removal costs the
        cell one molecule of ATP&mdash;one unit of precious energy. Indeed, cyclic reactions of this kind were
        originally labeled &ldquo;futile.&rdquo; But the adjective is misleading. The addition of phosphate groups to
        proteins is the single most common reaction in cells and underpins a large proportion of the computations they
        perform. Far from being futile, this cyclic reaction provides the cell with an essential resource: a flexible
        and rapidly tunable device.</span>
    </p>
    <p> If the cell really needs to change the concentration of the modified protein very quickly, it can. All it has to
      do is to switch on or shut off the phosphate-adding reaction and the concentration will fall
      precipitously&mdash;at the speed of the spinning cycle. There is no buildup of products or depletion of substrates
      to slow down the process, as there would be in a linear chain of enzyme reactions. </p>
    <p>
      <span>Regulation via phosphorylation and dephosphorylation (by &ldquo;kinases&rdquo; and
        &ldquo;phosphatases&rdquo; respectively) is an extremely general feature of life. &ldquo;About 30&ndash;50% of
        human proteins contain covalently attached phosphate. [. . .] A typical mammalian cell makes use of hundreds of
        distinct types of protein kinases at any moment.&rdquo; [Alberts]</span>
    </p>
    <p>
      <span>Down the rabbit-hole&hellip;</span>
    </p>
    <p>
      <span>One thing I don&rsquo;t love in presentations of chemotaxis&mdash;and of biological concepts
        generally&mdash;is that they often prominently feature flowcharts and network diagrams. In the case of
        chemotaxis, as you can gather from the video above, there are many players with nearly indistinguishable names:
        CheA phosphorylates CheY to become CheY-p, and CheZ dephosphorylates it back to CheY; CheW couples CheA to the
        receptors, and CheR methylates those receptors&rsquo; struts; CheB, meanwhile, &ldquo;clips off&rdquo; the
        methyl groups added to the struts by CheR. A network diagram is no doubt useful for organizing this sea of names
        but in a sense it foregrounds the most abstract view of the process. I&rsquo;d rather try to get a sense of the
        parts as a living whole or in their individual physical detail.</span>
    </p>
    <p>
      <span>When you do that, it&rsquo;s amazing what you find.</span>
    </p>
    <p>
      <span>What does it mean for a receptor to detect attractant?<br /></span><span>Almost every action in a cell
        depends on proteins changing shape and binding to each other. It&rsquo;s no different in the </span><span>E.
        coli</span><span>&nbsp;receptor complex.</span>
    </p>
    <p>
      <span>The way it works is that there are stimulus-specific proteins floating just outside the </span><span>E.
        coli</span><span>&rsquo;s cell membrane, in what&rsquo;s known as the periplasm. These proteins are
        &ldquo;stimulus-specific&rdquo; in the literal sense that they are shaped so as to bind favorably&mdash;in fact
        exclusively&mdash;with individual molecules of attractant. </span><span>E. coli</span><span>&nbsp;has five or
        six of these, for instance one that detects a crucial amino acid called aspartate. This sensor protein has
        little clefts in it that are shaped just so for molecules of aspartate to fit snugly into them. [Falke] In
        schematic form the aspartate receptor looks like this:</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image1.png" />
    </p>
    <p>
      <span>(Source: Falke, </span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2899694/&amp;sa=D&amp;source=editors&amp;ust=1691027434685513&amp;usg=AOvVaw074L-N5yK-vvXTnLuWgjtq">&ldquo;THE
          TWO-COMPONENT SIGNALING PATHWAY OF BACTERIAL CHEMOTAXIS: A Molecular View of Signal Transduction by Receptors,
          Kinases, and Adaptation Enzymes&rdquo;</a></span><span>)</span>
    </p>
    <p>
      <span>You can see that the sensory part&mdash;up top, where the aspartate binds&mdash;is connected to the
        signaling proteins CheW and CheA by a columnar structure that straddles the cell&rsquo;s membrane. What does
        this protein complex &ldquo;actually&rdquo; look like?</span>
    </p>
    <p>
      <span>An individual protein is small enough&mdash;</span><span><a
          href="https://www.google.com/url?q=http://book.bionumbers.org/how-big-is-the-average-protein/&amp;sa=D&amp;source=editors&amp;ust=1691027434686038&amp;usg=AOvVaw0DBI02dpEsa2MxHF-VJGOv">like
          a few tens of nanometers wide</a></span><span>&mdash;that it can&rsquo;t really be seen through a light
        microscope. This receptor from top to bottom measures about 350 angstroms, or 35 nanometers. But modern biology
        is all about seeing the unseeable. Nowadays, we try to find out what nanostructures look like by </span><span><a
          href="https://www.google.com/url?q=https://en.wikipedia.org/wiki/X-ray_crystallography&amp;sa=D&amp;source=editors&amp;ust=1691027434686323&amp;usg=AOvVaw3V_JjYIEHIIPWBQf2saUd0">X-ray
          diffraction</a></span><span>&nbsp;or, more and more often, by cryo-freezing them in an electron microscope.
        Once we determine a protein&rsquo;s structure it&rsquo;s usually </span><span>rendered using ribbon diagrams, a
        style </span><span><a
          href="https://www.google.com/url?q=https://stories.duke.edu/sciences-mother-of-ribbon-diagrams-celebrates-50-years-at-duke&amp;sa=D&amp;source=editors&amp;ust=1691027434686636&amp;usg=AOvVaw3wQFiELtJ8ppAnMsoRUeUM">invented
          by the biochemist Jane Richardson</a></span><span>&nbsp;in the late 1970s. Here&rsquo;s a ribbon diagram for
        the </span><span>E. coli</span><span>&nbsp;receptor:</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image10.png" />
    </p>
    <p>
      <span>(Source: </span><span><a
          href="https://www.google.com/url?q=https://ckcassidycom.files.wordpress.com/2015/12/full_trimer.jpg&amp;sa=D&amp;source=editors&amp;ust=1691027434687247&amp;usg=AOvVaw1LTK1OWPBiCSEfk1nF1xk9">Keith
          Cassidy</a></span><span>)</span>
    </p>
    <p>
      <span>This whole thing is the receptor. (Those parts just inside the membrane, with little yellow methyl groups
        lingering stuck to them, are the struts.) In effect it acts like one big piston: what happens is that when the
        asparate binds to the part in the periplasm, the columnar structure it&rsquo;s attached to changes
        shape&mdash;it &ldquo;dips&rdquo; and &ldquo;tilts&rdquo;&mdash;in such a way to lock the thing that&rsquo;s
        supposed to be phosphorylating CheY, the CheA kinase, into an inactive state.</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image12.gif" />
    </p>
    <p>
      <span>(Adapted from: </span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3DMCobqYrE67w&amp;sa=D&amp;source=editors&amp;ust=1691027434687865&amp;usg=AOvVaw2oDxYf-BTq5azog22itNZD">CheA
          conformation change, posted by Keith Cassidy</a></span><span>)</span>
    </p>
    <p>
      <span>When I think of a cell I imagine a Rube Goldberg&ndash;type contraption where an arm swings here, which
        drops a ball into a slide there, which rolls down and opens a trap door, which... eventually turns on or off
        some important cellular function. Indeed, </span><span>E. coli</span><span>&rsquo;s &ldquo;sense of smell&rdquo;
        rests ultimately in a series of physical lock-and-key mechanisms, starting with literal molecules of e.g.
        aspartate nuzzling into a protein and transmitting that physical shape-change across the membrane.</span>
    </p>
    <p>
      <span>This piston-shaped receptor protein is just one of a huge array arranged near the front of </span><span>E.
        coli</span><span>&rsquo;s body. In cross section they appear almost to have been laid down through a lithography
        process, in a neat hexagonal pattern:</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image2.png" />
    </p>
    <p>
      <span>(Source: </span><span><a
          href="https://www.google.com/url?q=https://www.ks.uiuc.edu/Research/chemotaxis/&amp;sa=D&amp;source=editors&amp;ust=1691027434688829&amp;usg=AOvVaw3ckwBIcaiSonRZlv1pAHdi">Theoretical
          and Computational Biophysics Group, UIUC</a></span><span>)</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image13.png" />
    </p>
    <p>
      <span>(Source: </span><span><a
          href="https://www.google.com/url?q=https://www.pnas.org/doi/10.1073/pnas.0610106104&amp;sa=D&amp;source=editors&amp;ust=1691027434689288&amp;usg=AOvVaw3JYUGruOQMeEjpsPl6G_Pc">Direct
          visualization of Escherichia coli chemotaxis receptor arrays using cryo-electron
          microscopy</a></span><span>)</span>
    </p>
    <p>
      <span>Calling </span><span>E. coli</span><span>&rsquo;s receptor complex its &ldquo;nose&rdquo; is no mere
        metaphor. Our own noses operate on a similar principle: when you smell a flower, it means that actual
        flower-molecules&mdash;possibly only a tiny number of them&mdash;have reached the inside of your nose and bonded
        to some protein with a specific affinity for that very molecule. This signal is then transmitted via nerves to
        your brain. The human nose has several hundred receptor proteins for smell; a dog has </span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5884888/%23:~:text%3DThe%2520canine%2520OR%2520repertoire%2520is%2520composed%2520of%25201%252C094%2520genes%252C%2520approximately%2520three%2520times%2520more%2520than%2520a%2520human.&amp;sa=D&amp;source=editors&amp;ust=1691027434689839&amp;usg=AOvVaw1vZzVpUdZsiyjH1-Kw71Yc">more
          than a thousand</a></span><span>. Every one of our senses works like this. Touch is underwritten by proteins
        that get &ldquo;squished&rdquo; by tactile forces into cell membranes, triggering a set of downstream responses.
        Sight is my favorite example. There&rsquo;s a protein called opsin that lives in the cells of our retina.
        What&rsquo;s so cool about it is that the thing that changes </span><span>its</span><span>&nbsp;shape is a
        literal photon. That is, </span><span><a
          href="https://www.google.com/url?q=https://pdb101.rcsb.org/motm/147%23:~:text%3DRhodopsin%252C%2520shown%2520here,to%2520the%2520brain.&amp;sa=D&amp;source=editors&amp;ust=1691027434690199&amp;usg=AOvVaw0oydktbZjODkkYA4U9JvuA">opsin
          converts the electromagnetic force of an incoming photon into a biomechanical / biochemical
          signal</a></span><span>. This is why I tend to think of molecular biology as the science of shapes bumping
        into each other.</span>
    </p>
    <p>
      <span>I think of </span><span>E. coli</span><span>&rsquo;s receptor complex as a protoversion of our own sensory
        apparatus. Its nose has only </span><span><a
          href="https://www.google.com/url?q=https://youtu.be/cT855rpX8bc?t%3D2596&amp;sa=D&amp;source=editors&amp;ust=1691027434690614&amp;usg=AOvVaw26f8X3yAyVG5QkwhryqmiD">five
          or six</a></span><span>&nbsp;attractant-specific sensory proteins, but their signals are integrated&mdash;as
        if different sets of receptor-protein activations were playing different &ldquo;chords&rdquo; on the
      </span><span>E. coli</span><span>&rsquo;s sensorium. &ldquo;In short, the chemosensory array functions as an
        ultrasensitive, ultrastable biological integrated circuit or sensory chip.&rdquo; [Falke 2]</span>
    </p>
    <p>
      <span>How the signal is carried<br /></span><span>So a bit of attractant binds one of the receptors, and lo, the
        equilibrium inside the cell begins to shift. Because the CheA kinase is now active, CheYs start getting
        phosphorylated much faster than they&rsquo;re </span><span>de</span><span>phosphorylated. Recall that this is a
        response that is dynamic, a flow that is tuned. The CheY-p concentration builds up. And then what?</span>
    </p>
    <p>
      <span>A CheY-p is just a CheY with a little kink in it, caused by the phosphate group now bound to it. But this
        kink has a dramatic effect on its behavior. It now loses its affinity for CheA (the thing that activated it) and
      </span><span>gains</span><span>&nbsp;an affinity for a protein at the base of the flagellar motor. It ends up
        binding to that, which&mdash;we&rsquo;ll see&mdash;flips the motor&rsquo;s rotation, etc., etc.</span>
    </p>
    <p>
      <span>But there&rsquo;s something really important worth dwelling on here. When we say that CheY-p has an
        &ldquo;affinity&rdquo; for the motor protein, it&rsquo;s not like it gets directed there; nor does it have some
        long-acting magnetic attraction for it. What this really means is that it has a strong inclination to bind to
        the motor protein when it gets </span><span>really really</span><span>&nbsp;close to it. Given how small a given
        CheY-p is in the scheme of the whole cell&rsquo;s cytoplasm, it might seem improbable that it&rsquo;ll somehow
        sidle up right next to one of these motor proteins somewhere on the other end of the </span><span>E.
        coli</span><span>&rsquo;s body. But that gets at the heart of </span><span><a
          href="https://www.google.com/url?q=https://www.newyorker.com/magazine/2022/03/07/a-journey-to-the-center-of-our-cells&amp;sa=D&amp;source=editors&amp;ust=1691027434691534&amp;usg=AOvVaw16R74wQuGR0a8MaIoM1aAD">the
          crazy kinetic chaos inside our cells</a></span><span>.</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image19.png" />
      <img loading="lazy" alt="" src="articleImages/image7.jpg" />
      <span>&nbsp; <br />(Source: David Goodsell, </span><span>The Machinery of Life</span><span>)</span>
    </p>
    <p>
      <span>Cells are dense with stuff, but everything in it is also
      </span><span>extremely</span><span>&nbsp;fast-moving:</span>
    </p>
    <p> To get an idea of how fast this motion is, imagine a typical bacterial cell, and place an enzyme at one end and
      a sugar molecule at the other. They will bump around and wander through the whole cell, encountering many
      molecules along the way. On average, though, it will only take about a second for those two molecules to bump into
      each other at least once. This is truly remarkable: this means that any molecule in a typical bacterial cell,
      during its chaotic journey through the cell, will encounter almost every other molecule in a matter of seconds.
      [Goodsell] </p>
    <p>
      <span>Just to put this in perspective: imagine you took an </span><span>E. coli</span><span>&nbsp;cell and scaled
        it up so that it was the length of a football field. And imagine you kept all the physics the same. A water
        molecule would be about an inch wide; a protein would be about the size of a basketball. The proteins would be
        juddering violently due to the thermal motion of the water particles bombarding them&mdash;so violently in fact
        that if left unchecked they&rsquo;d be moving at </span><span>500 meters per second</span><span>. But they
        aren&rsquo;t left unchecked: if you were in such an environment it would be so crowded as to be nearly
        impossible to see. What you really get, then, is an incredible ceaseless shaking and bouncing-into-each-other of
        all the component parts.</span>
    </p>
    <p>
      <span>This is why shape changes that lead to different bonding affinities are so important in biology. It&rsquo;s
        as if inside a cell everyone is constantly going up to everyone else, seeing if they fit together. Proteins
        sample the space of interactions with other proteins so quickly that for a long time, most biologists
        didn&rsquo;t really contemplate </span><span>where</span><span>&nbsp;in the cytoplasm two reactants lived; they
        knew that you never had to wait too long for them to meet each other. In fact it was a relatively recent
        discovery that inside the cytoplasm certain proteins that share functional relationships
      </span><span>do</span><span>&nbsp;seem to keep especially close together, inside little oil drops known as
        &ldquo;phase-separated liquids.&rdquo; Weak interactive forces between the floppy tails of different proteins
        cause them to spontaneously &ldquo;phase separate&rdquo; into these more viscous pools, and this biases them to
        interact more frequently.</span>
    </p>
    <p>
      <span>The rate-limiting step in </span><span>E. coli</span><span>&rsquo;s reaction to attractant is the time it
        takes for CheY-p to diffuse from the nose to the motor. It takes about a tenth of a second. The journey has
        actually been tracked on camera, using a fluorescent version of the protein:</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image17.png" />
    </p>
    <p>
      <span>(Source: </span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5052738/&amp;sa=D&amp;source=editors&amp;ust=1691027434693001&amp;usg=AOvVaw0Jxwg3f7VpZZub1RlYzT2d">Single-molecule
          imaging of electroporated dye-labelled CheY in live Escherichia coli</a></span><span>)</span>
    </p>
    <p>
      <span>What happens when the signal reaches the motor?<br /></span><span>Let&rsquo;s talk about these motors. These
        things are so intricate and beautiful and seem so reminiscent of machines we&rsquo;d engineer ourselves that
        they&rsquo;re sometimes cited as evidence for intelligent design.</span>
    </p>
    <p>
      <span>The flagellar motor operates with nearly 100% energy efficiency, near the limit of thermal noise. It spins
        at about 1,500 rotations per second. And the craziest part is that like all molecular nanomachines it is
        entirely self-assembled. There&rsquo;s an amazing 30-minute documentary </span><span><a
          href="https://www.google.com/url?q=https://youtu.be/uw0-MHI_248&amp;sa=D&amp;source=editors&amp;ust=1691027434693464&amp;usg=AOvVaw3PbbZysNNBqQLlFSAWmRSv">available
          on YouTube</a></span><span>&nbsp;that details the mechanics of the self-assembly process&mdash;and,
        refreshingly, profiles some of the scientists who figured it out, describing the methods they used to make their
        discoveries.</span>
    </p>
    <p>
      <span>(Source: </span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3Duw0-MHI_248&amp;sa=D&amp;source=editors&amp;ust=1691027434693881&amp;usg=AOvVaw3h7SQOMSvrU6cElvJLGtu-">Self-Assembling
          NanoMachine: a film about flagellar biosynthesis</a></span><span>)</span>
    </p>
    <p>
      <span>My favorite part of the self-assembly process is that after building a base for the rotor, a sort of tunnel
        is built and the proteins that comprise the whip-like &ldquo;hook&rdquo; of the flagellum are extruded through
        it&mdash;as if the flagellum were built by vomiting forth parts of itself.</span>
    </p>
    <p>
      <span>Anyway, at the base of each rotor there are a series of proteins called FliG, FliM, and
        FliN&mdash;pronounced like &ldquo;Fly G,&rdquo; &ldquo;Fly M,&rdquo; &ldquo;Fly N&rdquo;&mdash;to which CheY-p,
        our Frodo-esque bearer of the message from the nose, attaches once it finally arrives. CheY-p has a strong
        affinity for FliG and will readily glom onto it. We&rsquo;ll see how that actually affects the flagellum in a
        second. But for now it&rsquo;s worth noting that there&rsquo;s a thresholding mechanism here: just one CheY-p
        attaching to FliG won&rsquo;t be enough to flip the motor from counter-clockwise to clockwise (thereby causing a
        tumble)&mdash;it actually takes a handful of CheY-ps conspiring to make that happen. In fact the motor has
        something like seven states, from rotating quickly counterclockwise at three discrete levels of decreasing
        speed&mdash;as if stepping through three gears on a bike&mdash;to stalling entirely, to starting back up again
        in the clockwise direction, also with three speeds.</span>
    </p>
    <p>
      <span>Even as the motor is in the process of changing direction, any CheY-ps that
      </span><span>do</span><span>&nbsp;attach to FliG are under constant threat of being removed by yet another player,
        CheZ. That is, the proteins that would reverse the motors are subject to removal by other proteins that
      </span><span>un</span><span>-reverse it. Again we have a responsive regulatory circuit reminiscent of the one that
        phosphorylated and de-phosphorylated CheY in the first place upstream at the receptor. The idea is that every
        effect is reversible, and in fact is reversed at a regular rate. This means that in the absence of further
        signal the cell will quickly return to baseline.</span>
    </p>
    <p>
      <span>How does the motor actually change directions?<br /></span><span>As a matter of pure mechanics this might be
        the most ingenious part of the story. It took quite a long time to figure out and even still it seems that
        we&rsquo;re not entirely confident with our explanation. But the mechanism that&rsquo;s been proposed is that
        CheY-p binds to a protein called FliM (&ldquo;Fly EM&rdquo;) embedded in that ring that defines the base of the
        rotor. This tilts it and causes a 90-degree rotation in an attached protein called
        FliG</span><span>c</span><span>. That protein sits at the interface between the rotating part of the motor and
        the so-called &ldquo;stator,&rdquo; which drives it from the part that&rsquo;s anchored solidly in the cell
        membrane. </span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image11.jpg" />
    </p>
    <p>
      <span>(Source: </span><span><a
          href="https://www.google.com/url?q=https://www.pnas.org/doi/10.1073/pnas.1110111108&amp;sa=D&amp;source=editors&amp;ust=1691027434695078&amp;usg=AOvVaw3HahrdKGLlkUOqDyVdmoNO">A
          molecular mechanism of direction switching in the flagellar motor of Escherichia coli</a></span><span>)</span>
    </p>
    <p>
      <span>When FliG</span><span>c </span><span>changes orientation, the stepper-motor-like cycle that normally drives
        the motor counterclockwise starts driving it clockwise instead. In the illustration below, Figure A shows the
        stator, i.e., the driving mechanism of the motor. It works by stepping back and forth between the
        &ldquo;open&rdquo; and &ldquo;closed&rdquo; states, schematized by the </span>
      <img loading="lazy" alt="" src="articleImages/image14.png" />
      <span>and </span>
      <img loading="lazy" alt="" src="articleImages/image3.png" />
      <span>symbols respectively. In Figure B you can see how, in the normal CCW direction, the repeated cycling between
        these two states drives the &ldquo;teeth&rdquo; of the motor&mdash;the crucial FliG</span><span>c
      </span><span>proteins, here tilted left-to-right.</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image9.png" />
    </p>
    <p>
      <span>(Source: Supplementary figure S9, </span><span><a
          href="https://www.google.com/url?q=https://www.pnas.org/doi/10.1073/pnas.1110111108&amp;sa=D&amp;source=editors&amp;ust=1691027434695765&amp;usg=AOvVaw3RReCve7XbneGtbbm3RVo9">A
          molecular mechanism of direction switching in the flagellar motor of Escherichia coli</a></span><span>)</span>
    </p>
    <p>
      <span>When the CheY-p arrives at the rotor it has the effect of flipping the FliG</span><span>c
      </span><span>proteins so that now they tilt right-to-left. In that orientation the step-drive action works the
        opposite way, and the motor rotates clockwise:</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image6.png" />
    </p>
    <p>
      <span>You can see this more clearly below in this animated version of those figures. The animation starts by
        showing the clockwise motion; dwells on the orientation change of the FliG</span><span>c
      </span><span>&nbsp;proteins; then finally continues roating counterclockwise.</span>
    </p>
    <p>
      <span style="
            overflow: hidden;
            display: inline-block;
            margin: 0px 0px;
            border: 0px solid #000000;
            transform: rotate(0rad) translateZ(0px);
            -webkit-transform: rotate(0rad) translateZ(0px);
            width: 307.9px;
            height: 300.74px;
          "><img loading="lazy" alt="" src="articleImages/image15.gif" style="
              width: 307.9px;
              height: 300.74px;
              margin-left: 0px;
              margin-top: 0px;
              transform: rotate(0rad) translateZ(0px);
              -webkit-transform: rotate(0rad) translateZ(0px);
            " title="" /></span>
    </p>
    <p>
      <span>It&rsquo;s a shame that the paper presenting this theory didn&rsquo;t include such an animation, because
        that by far is the most natural way to consume this information. It speaks to the fact that creating animations
        is really hard! (It took several hours even to produce this crude version.) Per Bret Victor&rsquo;s talk
      </span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3DZfytHvgHybA&amp;sa=D&amp;source=editors&amp;ust=1691027434696583&amp;usg=AOvVaw0Dwf4h00T3P3NxSdEEQAN9">Stop
          Drawing Dead Fish</a></span><span>, making moving pictures shouldn&rsquo;t be so difficult. If it were easier,
        such animations would spread everywhere in scientific communication, because so often what a paper describes is
        some kind of dynamic process. Dynamic illustrations would help readers grasp proposed mechanisms more quickly.
        As it is, someone who understands a complex mechanism usually has to explain it, in patient detail, to someone
        who&rsquo;s good at animating; this costs time and money; and most people simply opt not to go through with it.
        Perhaps, someday, the process will be democratized by better tools, or a multimodal AI system.</span>
    </p>
    <p>
      <span>How the motor changing directions causes the E. coli to tumble<br /></span><span>The final part of the
        story&mdash;for me, anyway; there&rsquo;s a lot more to explore!&mdash;is why exactly the clockwise rotation of
        just one of the flagellar motors would send the whole cell a-tumbling. It helps to understand how the thing
        works in &ldquo;run&rdquo; mode, when all the flagella are oriented the same way.</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image4.png" />
    </p>
    <p>
      <span>(Traced from: </span><span><a
          href="https://www.google.com/url?q=https://www.nsf.gov/news/mmg/mmg_disp.jsp?med_id%3D66718%26from%3D&amp;sa=D&amp;source=editors&amp;ust=1691027434697127&amp;usg=AOvVaw3HQpLM1prOeict1SBHHrge">How
        </a></span><span><a
          href="https://www.google.com/url?q=https://www.nsf.gov/news/mmg/mmg_disp.jsp?med_id%3D66718%26from%3D&amp;sa=D&amp;source=editors&amp;ust=1691027434697331&amp;usg=AOvVaw3zc0_IEAU_ymDG1MxyY-8Z">Escherichia
          coli</a></span><span><a
          href="https://www.google.com/url?q=https://www.nsf.gov/news/mmg/mmg_disp.jsp?med_id%3D66718%26from%3D&amp;sa=D&amp;source=editors&amp;ust=1691027434697537&amp;usg=AOvVaw1RY7Cwu7mzmjgANxg1hS-5">&nbsp;Move
          | National Science Foundation</a></span><span>)</span>
    </p>
    <p>
      <span>Even though this bundle of flagella sort of looks like a propeller, when you actually think about it,
        that&rsquo;s not really what it is. It&rsquo;s more like a pig&rsquo;s curly tail that spins with a whip-y sort
        of motion. How exactly does that propel the entire cell? A wonderful book called </span><span>Random Walks in
        Biology</span><span>&nbsp;gets into the physics in some detail:</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image18.png" />
    </p>
    <p>
      <span>(Source: Howard C. Berg, </span><span>Random Walks in Biology</span><span>)</span>
    </p>
    <p>
      <span>To get a grip on things like this, it helps to have a model of some kind that you can hold in your hand. And
        actually the question of how separate filaments running in phase near each other would come to bundle was
        explored nicely in </span><span><a
          href="https://www.google.com/url?q=https://www.pnas.org/doi/10.1073/pnas.2633596100&amp;sa=D&amp;source=editors&amp;ust=1691027434698224&amp;usg=AOvVaw2q1vq2A0-ckah_Z2ZJGYk0">this
          paper</a></span><span>.</span><span>&nbsp;The authors used physical models of the flagella by wrapping hollow
        Tygon tubes around a mandrel and filling them with epoxy. They then used a couple stepper motors to drive the
        counter-clockwise rotation. &ldquo;The flow field generated by each helix tilts the other helix, causing the
        helices to roll around each other and form a right-handed wrapping&rdquo;:</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image8.gif" />
    </p>
    <p>
      <span>(Source: </span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3D25FtMdIFtXM&amp;sa=D&amp;source=editors&amp;ust=1691027434698566&amp;usg=AOvVaw0Z-kBaMs-wmT_Vag4PrsjD">Macro
          Scale Model E.Coli Flagella Bundling</a></span><span>)</span>
    </p>
    <p>
      <span>Individuality in the bacterial population<br /></span><span>We tend to think of a colony of something like
      </span><span>E. coli</span><span>&nbsp;as an undifferentiated evil goo, each bacterium identical to its neighbors.
        But people who&rsquo;ve studied these organisms under the microscope observe a surprising amount of individual
        personality. A 1976 </span><span>Nature</span><span>&nbsp;</span><span><a
          href="https://www.google.com/url?q=https://www.nature.com/articles/262467a0&amp;sa=D&amp;source=editors&amp;ust=1691027434698982&amp;usg=AOvVaw31jX-dH8cluf1gxfj2MP0b">paper</a></span><span>,
        &ldquo;Non-genetic individuality: chance in the single cell,&rdquo; explores variation in the context of
        chemotaxis using strains of Salmonella and Enterobacter bacteria.</span>
    </p>
    <p>
      <span>The paper came out before the exact mechanism behind chemotactic regulation was well-understood; all the
        authors knew was that &ldquo;control of tumbling can be rationalised as caused by changes in the levels of a
        tumble regulator.&rdquo; They hypothesized that although bacteria of the same strain would all share the exact
        same DNA, there might be a relatively small number of copies of that &ldquo;tumble regulator,&rdquo; and natural
        variation in the transcription, translation, and destruction of these regulator proteins could account for
        differences in behavior. Their experiments were mostly at the behavioral level. They observed how different
        individuals&mdash;including those in a particularly &ldquo;tumbly&rdquo; mutant strain (I love that
        word)&mdash;reacted to environments with and without attractant, and found plenty of variance.</span>
    </p>
    <p>
      <span>Their theory was spot-on. We talked above about how </span><span>E. coli</span><span>&nbsp;adapt to higher
        and higher concentrations of attractant via a clever methylation mechanism. Well, it turns out that the
        methylation of the receptor struts is governed by only about </span><span><a
          href="https://www.google.com/url?q=https://youtu.be/cT855rpX8bc?t%3D3032&amp;sa=D&amp;source=editors&amp;ust=1691027434699445&amp;usg=AOvVaw1xaIZzQ4eomN6Tl_uXFPRc">100
          CheR proteins</a></span><span>&nbsp;in the cell. The number of those proteins&mdash;along with CheB, which
      </span><span>un</span><span>-methylates the struts&mdash;determines the speed of the &ldquo;futile cycle&rdquo;
        that reacts to changes in attractant concentration. That is, it affects how quickly the bacterium adapts when
        the concentration goes up and refracts when it goes down. [Gore </span><span><a
          href="https://www.google.com/url?q=https://youtu.be/cT855rpX8bc?t%3D3989&amp;sa=D&amp;source=editors&amp;ust=1691027434699717&amp;usg=AOvVaw2M-YWjQP47EHmki6bhq4lU">1:06:30</a></span><span>]
        Because 100 copies of CheR is so extraordinarily tiny in the context of the full cell buzzing with something
        like ten million proteins, variation by just a handful can have a relatively large effect on the cell&rsquo;s
        behavior. [Gore </span><span><a
          href="https://www.google.com/url?q=https://youtu.be/cT855rpX8bc?t%3D4390&amp;sa=D&amp;source=editors&amp;ust=1691027434699977&amp;usg=AOvVaw3_amNW6qlWAsmTwSBtptRi">1:13:20</a></span><span>]
        That helps account for why different </span><span>E. coli</span><span>&nbsp;with the exact same genetic sequence
        will tumble and adapt at different frequencies.</span>
    </p>
    <p>
      <span>How did we figure all this stuff out?</span><span><br />We don&rsquo;t yet have the technology to just
        observe all of the activity inside a living cell. That Goodsell painting above that shows the crowded cytoplasm
        packed with proteins is an artistic composite&mdash;backed by rigorous research to be sure&mdash;because
        there&rsquo;s no way to capture all the different players in situ at once. And obviously it&rsquo;s a
        &ldquo;still life,&rdquo; not a video. So how could we possibly know all this detail about what exactly a given
        protein looks like, and how and when it interacts with others to kick off some particular part of the chemotaxis
        process?</span>
    </p>
    <p>
      <span>There seem to be three or four major kinds of experiment. Probably the most important is genetic: you can
        selectively disrupt one gene at a time and, by observing how the mutant </span><span>E.
        coli</span><span>&nbsp;behaves, begin to get a grip on each gene&rsquo;s function. All of the proteins
        &ldquo;CheY,&rdquo; &ldquo;CheZ,&rdquo; &ldquo;CheW,&rdquo; and so on are named simply because they are the
        products of genes that, when excised, &ldquo;cause a general defect in chemotaxis.&rdquo; [How Bacteria Sense
        and Swim] As you can imagine, identifying all of these is painstaking work, and involves a considerable amount
        of clever inference. For instance you might observe that without gene </span><span>X</span><span>&nbsp;the
        bacteria never seems to tumble; is that because that gene is involved in recognizing attractant or in forcing
        the rotor to run clockwise?</span>
    </p>
    <p>
      <span>Once you have a hypothesis, a second kind of experiment involves purifying some subset of these
        proteins-of-interest in vitro to see how they work together to form a particular signaling pathway. For example
        you could put CheA and CheY along with some phosphate groups and other necessary reactants and observe whether
        and how much phosphorylation takes place. That&rsquo;s what the authors did in </span><span><a
          href="https://www.google.com/url?q=https://doi.org/10.1016/0092-8674(90)90429-I&amp;sa=D&amp;source=editors&amp;ust=1691027434700671&amp;usg=AOvVaw2ymdzkkDwuPdCahUcas_z0">this
          paper</a></span><span>&nbsp;in </span><span>Cell</span><span>, in 1990. They used a radioactive version of
        phosphate as a tracer. &ldquo;Incorporation of [32P]phosphate into CheA or CheY was determined by excising the
        radioactive band out of the dried gel and quantitating in scintillation fluid or by analysis of the intact gel
        using a Phosphorimager (Molecular Dynamics, Sunnyvale, CA) and compare with known radioactive standards.&rdquo;
        Another common method for observing in vitro dynamics is to genetically modify proteins to fluoresce; or to
        &ldquo;find&rdquo; a protein in solution using an antibody that recognizes some part of it&mdash;you attach that
        antibody to </span><span>another</span><span>&nbsp;protein, and that one you fluoresce, so you can find the
        hidden one.</span>
    </p>
    <p>
      <span>To understand the literal lock-and-key mechanics at a particular binding site&mdash;for instance how exactly
        a molecule of aspartate causes a receptor to deform, kicking off a signaling cascade&mdash;involves
        &ldquo;structural&rdquo; biology work, i.e., taking pictures of individual proteins or, increasingly, ensembles
        of them in situ. For this you can use X-ray crystallography, nuclear magnetic resonance imaging, cryo-electron
        microscopy, super-resolution light microscopy, or some combination.</span>
    </p>
    <p>
      <span>A </span><span><a
          href="https://www.google.com/url?q=https://www.ks.uiuc.edu/Research/chemotaxis/&amp;sa=D&amp;source=editors&amp;ust=1691027434701279&amp;usg=AOvVaw0R2tNPp9anOI9VYBnwWR_e">group</a></span><span>&nbsp;at
        University of Illinois at Urbana-Champagne uses atomic-scale molecular dynamics simulations, in software, to
        understand structural details&mdash;like the exact </span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3DMCobqYrE67w&amp;sa=D&amp;source=editors&amp;ust=1691027434701492&amp;usg=AOvVaw2rhSeppvZ9AIZjw1P7cRv_">way</a></span><span>&nbsp;that
        CheA changes shape to kick off a downstream signaling process&mdash;that wouldn&rsquo;t be apparent from
        high-resolution imaging alone.</span>
    </p>
    <p>
      <span>Sometimes you can&rsquo;t get a direct picture. It may require deduction to understand, say, how exactly a
        protein fits in. </span><span><a
          href="https://www.google.com/url?q=https://pubmed.ncbi.nlm.nih.gov/1326408/&amp;sa=D&amp;source=editors&amp;ust=1691027434701805&amp;usg=AOvVaw3Z6M5TRkW1F1zd8NfGg6hL">One
          experiment</a></span><span>&nbsp;found that CheA didn&rsquo;t bind to a receptor except in the presence of
        CheW; that plus the fact that adding too much CheW into the mixture actually led to a
      </span><span>decrease</span><span>&nbsp;in the ability of CheA&ndash;CheW complexes to bind receptor suggested
        that CheW competed with that complex for the binding site on the receptor and that therefore it must sit
      </span><span>between</span><span>&nbsp;the receptor and CheW in the receptor&ndash;CheW&ndash;CheA trimer [Cited
        in Blair]. </span><span>Biology is lousy with heroic inferences like that. It&rsquo;s a world that&rsquo;s hard
        to see; sometimes you just have to </span><span>imagine</span><span>&nbsp;what&rsquo;s going on down there, and
        back up those imaginings with the right experiments.</span>
    </p>
    <p>
      <span>The very idea that bacteria run and tumble came from experiments </span><span><a
          href="https://www.google.com/url?q=https://www.nature.com/articles/239500a0&amp;sa=D&amp;source=editors&amp;ust=1691027434702217&amp;usg=AOvVaw1nVepK9yFtlLGuC3bBG3Nk">published
          in 1972</a></span><span>&nbsp;by Howard Berg and Douglas Brown, who used a special three-dimensional tracking
        microscope of their own design to watch the little suckers in action. (A fun fact is that they called the
        non-runs &ldquo;twiddles&rdquo; instead of &ldquo;tumbles.&rdquo;) Some of the physics of flagellar
        propulsion&mdash;like how much force the little tails generate&mdash;</span><span><a
          href="https://www.google.com/url?q=https://www.google.com/books/edition/An_Introduction_to_Systems_Biology/tcxCkIxzCO4C?hl%3Den%26gbpv%3D1%26pg%3DPAfrontcover%26printsec%3Dfrontcover&amp;sa=D&amp;source=editors&amp;ust=1691027434702463&amp;usg=AOvVaw0vDchnSBj4ndosA4B4P-R7">was
          discovered</a></span><span>&nbsp;later by tethering the flagella to a microscope slide: you can then measure
        how fast the </span><span>E. coli</span><span>&rsquo;s body spins. We know that bacterial flagellar motors are
        powered by the </span><span><a
          href="https://www.google.com/url?q=https://link.springer.com/referenceworkentry/10.1007/978-3-642-11274-4_141&amp;sa=D&amp;source=editors&amp;ust=1691027434702769&amp;usg=AOvVaw3-hK3ao-_VgPQ3YRoBGffL">proton
          motive force</a></span><span>&nbsp;from a </span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC431412/&amp;sa=D&amp;source=editors&amp;ust=1691027434703011&amp;usg=AOvVaw0ZEge0h3PEEzfuajJv9C7G">1977
          paper</a></span><span>&nbsp;that measured how cells ran or &ldquo;twiddled&rdquo; in the presence or absence
        of an electrical potential. But the research has become even more refined than that. Just by observing the
        strength of the rotation under various conditions&mdash;different viscosities, temperatures, and so on&mdash;we
        know that &ldquo;rotation is tightly coupled to proton flow, with a fixed number of protons (~500) used to drive
        each revolution.&rdquo; [Blair] Think of how detailed an understanding we&rsquo;ve gotten!</span>
    </p>
    <p>
      <span>One reason I&rsquo;m particularly attracted to studies of </span><span>E. coli</span><span>&nbsp;chemotaxis
        is that it&rsquo;s an early star of what&rsquo;s been called &ldquo;in silico&rdquo; biology. It&rsquo;s been
        the subject of many computer models. Dennis Bray, the author of that book that put me onto this stuff in the
        first place, made one of the more </span><span><a
          href="https://www.google.com/url?q=https://pubmed.ncbi.nlm.nih.gov/17208180/&amp;sa=D&amp;source=editors&amp;ust=1691027434703393&amp;usg=AOvVaw3bo3Zcz6DZBqmlF3yVZKcg">popular
          models</a></span><span>. Here&rsquo;s a nice still shot of the model in action:</span>
    </p>
    <p>
      <img loading="lazy" alt="" src="articleImages/image16.jpg" />
    </p>
    <p>
      <span>Maybe the chief role of a computer model is that to get it working in the first place you have to explicitly
        articulate every one of your assumptions. In much the same way that writing tends to clarify your thinking (or
        at least reveal how unclear it really is), a computer model forces you to synthesize what you know. (Though if
        anything it&rsquo;s even more exacting than a blank page.)</span>
    </p>
    <p>
      <span>Once you have a model, you can use it to explore variations on those assumptions. &ldquo;The program gives
        the correct phenotype of over 60 mutants in which chemotaxis-pathway components are deleted or
        overexpressed,&rdquo; Bray writes. At best, a good enough model lets you discover things you didn&rsquo;t
        already know, or suggests your next experiment. &ldquo;In order to match the impulse response to a brief
        stimulus [. . .] we also had to increase the activities of the adaptational enzymes CheR and CheB at least an
        order of magnitude greater than published values.&rdquo;</span>
    </p>
    <p>
      <span>So what?<br /></span><span>Why should you care about </span><span>E. coli</span><span>&nbsp;chemotaxis? A
        typical answer to that sort of question&mdash;and I&rsquo;m sure the answer given in many of the grant
        applications supporting the work cited here&mdash;is that there are medical and practical uses. For instance: if
        you understand the signaling pathways of bacterial chemotaxis you can disrupt them; that work might lead to a
        new kind of antibiotic, which, in an era of increasing resistance, is direly needed. Or you might say that
        chemotaxis is used by the hunter-cells of your immune system to track down and clean up the cancerous, the dead,
        or the infected; understanding how it works can help make those cells work better, or disrupt their
        disrupters.</span>
    </p>
    <p>
      <span>More generally you might say&mdash;and in fact I led with this up top&mdash;that understanding this specific
        phenomenon equips you to understand all kinds of others. &ldquo;Bacterial two-component pathways
        [</span><span><a
          href="https://www.google.com/url?q=https://en.wikipedia.org/wiki/Two-component_regulatory_system&amp;sa=D&amp;source=editors&amp;ust=1691027434704156&amp;usg=AOvVaw0rtfe78sr95hPIrqAY1U8i">def&rsquo;n</a></span><span>]
        control a dazzling array of functions including cell division, virulence, antibiotic resistance, metabolite
        fixation and utilization, response to environmental stress, sporulation, and taxis.&rdquo;</span>
    </p>
    <p>
      <span>But I dunno, to me the real reason is that it&rsquo;s neat. It&rsquo;s just fun to find out about. &ldquo;To
        learn, and at due times to repeat what one has learnt, is that not after all a pleasure?&rdquo;</span>
    </p>
    <p>
      <span>Annotated bibliography<br /></span><span>Bray, Dennis. </span><span>Wetware: A Computer in Every Living
        Cell</span><span>. 2009. This is the book that kicked off my interest in E. coli chemotaxis. Bray built computer
        models of the network of interacting parts that give rise to E. coli&rsquo;s sophisticated foraging
        behavior&mdash;an early example of </span><span>in silico</span><span>&nbsp;&ldquo;systems biology.&rdquo; This
        is a very evocative book, giving you a newfound appreciation for the computational power of an individual
        cell.</span>
    </p>
    <p>
      <span>&ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3DLgPDOSou1tw&amp;sa=D&amp;source=editors&amp;ust=1691027434704780&amp;usg=AOvVaw3EyfzSaWGhCQu5Z1d0dx-a">Chemotaxis:
          Molecular Events</a></span><span>.&rdquo; A superb video that gives an overview of everything you need to know
        about chemotaxis. I think this was prepared for </span><span>Molecular Biology of the Cell</span><span>, the
        textbook cited below. Is it sufficient to just watch this video and be done with it? Maybe, but for some reason
        I had a bunch of questions I still wanted answered that led me to all this other stuff.</span>
    </p>
    <p>
      <span>Webre, Daniel J. et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.cell.com/current-biology/pdf/S0960-9822(02)01424-0.pdf&amp;sa=D&amp;source=editors&amp;ust=1691027434705166&amp;usg=AOvVaw0wgDM9vOgkXfa2vn8VXT1B">Bacterial
          chemotaxis</a></span><span>.&rdquo; Current Biology, </span><span><a
          href="https://www.google.com/url?q=https://www.sciencedirect.com/journal/current-biology/vol/13/issue/2&amp;sa=D&amp;source=editors&amp;ust=1691027434705365&amp;usg=AOvVaw2mWKbkJCFxNfqEtFec9hHx">Vol
          13 No 2</a></span><span>, 2003</span><span>. </span><span>Good short overview article, like an encyclopedia
        entry.</span>
    </p>
    <p>
      <span>Gore, Jeff. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3DcT855rpX8bc&amp;sa=D&amp;source=editors&amp;ust=1691027434705731&amp;usg=AOvVaw3YJA-chCV1TEZu1We1usOb">Robustness
          and Bacterial Chemotaxis.</a></span><span>&rdquo; 2014. This is a truly awesome lecture that gives you a sense
        of the majesty of bacterial chemotaxis and also of the weird physics of the microscopic world (which Gore calls
        &ldquo;Life at low Reynold&rsquo;s number&rdquo;). This lecture is part of Gore&rsquo;s </span><span><a
          href="https://www.google.com/url?q=https://ocw.mit.edu/courses/physics/8-591j-systems-biology-fall-2014/syllabus/&amp;sa=D&amp;source=editors&amp;ust=1691027434705987&amp;usg=AOvVaw2TVv5DgF3a2fkdLXzKJx10">MIT
          Systems Biology course</a></span><span>, available via MIT&rsquo;s amazing OpenCourseWare initiative.</span>
    </p>
    <p>
      <span>Alberts, Bruce et al. </span><span><a
          href="https://www.google.com/url?q=https://wwnorton.com/books/9780393884821&amp;sa=D&amp;source=editors&amp;ust=1691027434706220&amp;usg=AOvVaw0s8y4qVVfHbTb3d2AO9c0k">Molecular
          Biology of the Cell</a></span><span>, 7th edition, 2022. The meme bio textbook. It&rsquo;s good, and I
        actually in my adulthood paid for a digital copy of this thing. I refer to it every so often&mdash;actually
        maybe just twice since I paid for it. It feels handy, the kind of known-good and deep source it&rsquo;s helpful
        to have for answering your own and others&rsquo; basic questions. One thing that strikes me about this book,
        excellent as it is, is that it&rsquo;d probably be a tough way to actually
      </span><span>learn</span><span>&nbsp;biology. Instead, try the stuff cited in the &ldquo;Reading list&rdquo; at
        the bottom of </span><span><a
          href="https://www.google.com/url?q=https://jsomers.net/i-should-have-loved-biology/&amp;sa=D&amp;source=editors&amp;ust=1691027434706470&amp;usg=AOvVaw2fx9JvPUQahiuvmq4y19Oh">this
          post</a></span><span>.</span>
    </p>
    <p>
      <span>Falke, Joseph J. et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2899694/&amp;sa=D&amp;source=editors&amp;ust=1691027434706728&amp;usg=AOvVaw1K-D_E1q3P2hM7d1PjZYyY">The
          Two-Component Signaling Pathway of Bacterial Chemotaxis: A Molecular View of Signal Transduction by Receptors,
          Kinases, and Adaptation Enzymes.</a></span><span>&rdquo; 1997. A really nice paper that tells the basic story,
        if in somewhat forbidding technical language.</span>
    </p>
    <p>
      <span>Falke , Joseph J. et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4268382/&amp;sa=D&amp;source=editors&amp;ust=1691027434707010&amp;usg=AOvVaw264jhBafVCBxiK-olc2t5Y">Architecture
          and signal transduction mechanism of the bacterial chemosensory array: Progress, controversies, and
          challenges.</a></span><span>&rdquo; 2014. Another good review-ish paper.</span>
    </p>
    <p>
      <span>Bray, Dennis et al. </span><span><a
          href="https://www.google.com/url?q=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC300951/pdf/mbc00099-0029.pdf&amp;sa=D&amp;source=editors&amp;ust=1691027434707359&amp;usg=AOvVaw29bYFWHyFZ5Ylhd6rvi97Z">&ldquo;Computer
          Simulation of the Phosphorylation Cascade Controlling Bacterial Chemotaxis.&rdquo;</a></span><span>&nbsp;1993.
        A good example of </span><span>in silico</span><span>&nbsp;biology.</span>
    </p>
    <p>
      <span>Bray, Dennis et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.cell.com/current-biology/fulltext/S0960-9822(06)02508-5?_returnURL%3Dhttps%253A%252F%252Flinkinghub.elsevier.com%252Fretrieve%252Fpii%252FS0960982206025085%253Fshowall%253Dtrue&amp;sa=D&amp;source=editors&amp;ust=1691027434707847&amp;usg=AOvVaw2hSsgIJziAT7YtrbC1_2YM">The
          chemotactic behavior of computer-based surrogate bacteria.</a></span><span>&rdquo; 2007. Another, later paper
        by Bray with a more detailed computer model. </span>
    </p>
    <p>
      <span>&ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.ks.uiuc.edu/Research/chemotaxis/&amp;sa=D&amp;source=editors&amp;ust=1691027434708137&amp;usg=AOvVaw35fkesH8mdm4hTB24IgCum">Bacterial
          Chemotaxis</a></span><span>.&rdquo; A nice page from the Theoretical and Computational Biophysics group at the
        University of Illinois at Urbana&ndash;Champaign.</span>
    </p>
    <p>
      <span>Cassidy, C. Keith. </span><span><a
          href="https://www.google.com/url?q=https://ckcassidy.com/&amp;sa=D&amp;source=editors&amp;ust=1691027434708461&amp;usg=AOvVaw3F4ctQzRn3Gb6VuYiHWTIK">Personal
          website.</a></span><span>&nbsp;This fellow used to work at the UIUC group in the previous reference. He may be
      </span><span>the</span><span>&nbsp;expert on the chemotaxis signaling complex.</span>
    </p>
    <p>
      <span>Beniaguev, David et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.sciencedirect.com/science/article/pii/S0896627321005018&amp;sa=D&amp;source=editors&amp;ust=1691027434708819&amp;usg=AOvVaw3GBVnNnzYEh-W0AEigFfLc">Single
          cortical neurons as deep artificial neural networks.</a></span><span>&rdquo; 2021. A paper showing that
        &ldquo;cortical neurons are well approximated by a deep neural network (DNN) with 5&ndash;8
        layers.&rdquo;</span>
    </p>
    <p>
      <span>Koushik, Paul et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.pnas.org/doi/10.1073/pnas.1110111108&amp;sa=D&amp;source=editors&amp;ust=1691027434709153&amp;usg=AOvVaw3CClD6ecb5ojBdqRpeqq7U">A
          molecular mechanism of direction switching in the flagellar motor of Escherichia coli.</a></span><span>&rdquo;
        2011. This is where all those great diagrams showing a hypothesized mechanism for the rotor switching come from.
        I only wish they&rsquo;d animated them! (Maybe they felt that they were already out on a limb, and making a
        video of a proposed mechanism would make it seem realer than it deserve to be? Or maybe they just thought it was
        too hard.)</span>
    </p>
    <p>
      <span>Berg, Howard. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://web.archive.org/web/20220810172710/http://www.rowland.harvard.edu/labs/bacteria/movies/ecoli.php&amp;sa=D&amp;source=editors&amp;ust=1691027434709608&amp;usg=AOvVaw3_lpQUT6b_N-DJzJ6DPfHe">Swimming
        </a></span><span><a
          href="https://www.google.com/url?q=https://web.archive.org/web/20220810172710/http://www.rowland.harvard.edu/labs/bacteria/movies/ecoli.php&amp;sa=D&amp;source=editors&amp;ust=1691027434709842&amp;usg=AOvVaw1KQL2O1d80uzb5mHD2ZFpF">Escherichia
          coli</a></span><span><a
          href="https://www.google.com/url?q=https://web.archive.org/web/20220810172710/http://www.rowland.harvard.edu/labs/bacteria/movies/ecoli.php&amp;sa=D&amp;source=editors&amp;ust=1691027434710097&amp;usg=AOvVaw32qxXPHtBe0neViYlr_fyy">.</a></span><span>&rdquo;
        Archived web page with lots of videos of </span><span>E. coli</span><span>&nbsp;in motion. Berg was one of the
        OG chemotaxis people, and also wrote the amazing </span><span>Random Walks in Biology</span><span>, cited
        below.</span>
    </p>
    <p>
      <span>Alon, U. et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.nature.com/articles/16483&amp;sa=D&amp;source=editors&amp;ust=1691027434710533&amp;usg=AOvVaw05tmO-qLVFdd_bcxTZQBEg">Robustness
          in bacterial chemotaxis.</a></span><span>&rdquo; 1999. I don&rsquo;t know if I understood this paper all that
        well, except to get the point that the chemotaxis system doesn&rsquo;t rely on there being some precise
        concentration of any of the component parts, e.g. CheY. Like many biological systems it works in the face of
        lots of gotchas and what-have-yous.</span>
    </p>
    <p>
      <span>&ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3Duw0-MHI_248&amp;sa=D&amp;source=editors&amp;ust=1691027434710929&amp;usg=AOvVaw3JkgoWv9-uTBwVWFLHE7j5">Self-Assembling
          NanoMachine: a film about flagellar biosynthesis</a></span><span>.&rdquo; A documentary (!) about the way that
        the bacterial flagellar motor is made, and the various discoveries that elaborated the process. Pretty hard to
        believe that this documentary exists&mdash;but, there it is. It&rsquo;s awesome.</span>
    </p>
    <p>
      <span>Di Paolo, Diana et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://royalsocietypublishing.org/doi/10.1098/rstb.2015.0492&amp;sa=D&amp;source=editors&amp;ust=1691027434711292&amp;usg=AOvVaw2apea_KwgIEaxuK4VthsZQ">Single-molecule
          imaging of electroporated dye-labelled CheY in live E coli.</a></span><span>&rdquo; A near paper showing the
        tracking of CheY through the cytoplasm.</span>
    </p>
    <p>
      <span>MunJu, Kim et al. </span><span><a
          href="https://www.google.com/url?q=https://www.youtube.com/watch?v%3D25FtMdIFtXM&amp;sa=D&amp;source=editors&amp;ust=1691027434711608&amp;usg=AOvVaw3cJ3bKE3l7zpSyORTkEkKy">Macro
          Scale Model E. Coli Flagella Bundling</a></span><span>. YouTube video from </span><span><a
          href="https://www.google.com/url?q=https://www.pnas.org/doi/10.1073/pnas.2633596100&amp;sa=D&amp;source=editors&amp;ust=1691027434711811&amp;usg=AOvVaw1cRKEa9nYHuN2nHcs66XtE">this
          paper</a></span><span>. 2003. Uses the drill and tubing to simulate flagellar bundling.</span>
    </p>
    <p>
      <span>Sarkar, Mayukh K. et al. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://www.pnas.org/doi/10.1073/pnas.1000935107&amp;sa=D&amp;source=editors&amp;ust=1691027434712068&amp;usg=AOvVaw17ge-aF0tkq-KQn1NvYzY3">Chemotaxis
          signaling protein CheY binds to the rotor protein FliN to control the direction of flagellar rotation in
          Escherichia coli.</a></span><span>&rdquo; Pretty clear title!</span>
    </p>
    <p>
      <span>Vladimirov, Nikita. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://github.com/nvladimus/rapidcell&amp;sa=D&amp;source=editors&amp;ust=1691027434712301&amp;usg=AOvVaw0DsVUiDujqmgn6KpVsxlyZ">Multiscale
          modeling of E.coli chemotaxis.</a></span><span>&rdquo; Github project page. Here&rsquo;s a model I tried to
        actually run&mdash;and shockingly, it worked out of the box. Sadly it didn&rsquo;t show little virtual
      </span><span>E. colis</span><span>&nbsp;running and tumbling around on my screen, but still pretty cool. </span>
    </p>
    <p>
      <span>Blair, D F. &ldquo;</span><span><a
          href="https://www.google.com/url?q=https://pubmed.ncbi.nlm.nih.gov/8561469/&amp;sa=D&amp;source=editors&amp;ust=1691027434712611&amp;usg=AOvVaw05fX85kKr07HmgYyZkfCTY">How
          bacteria sense and swim.</a></span><span>&rdquo; 1995. A really well-written review article.</span>
    </p>
  </article>
  <div id="debug1"></div>
  <div id="debug2"></div>
  <script type="module" src="ecoliSimulation/index.js"></script>
</body>

</html>